### PR TITLE
Idle loop

### PR DESCRIPTION
**Pull request by aashishc1988**
Adding example for Idle loop.

The current example added for this module is wrong, i.e., I think we havent created an example for this and right now it is pointing to some other example.
Please see,
https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/api/platform/IdleLoop.md

### CODE REVIEW

**Comment by aashishc1988** [comment]
@AnotherButler

File: IdleLoop/main.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    // target-specific feed function
+};
+
+void watchdogRefreshIsr() {
+    watchdogQueue.call(callback(&watchdogRefreshHandler));
+};
+
+void new_idle_loop()
+{
+	led1 = !led1;
```
Comment (cmonr) [inline IdleLoop/main.cpp]: Intendation

File: IdleLoop/main.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+int main() {
+	watchdogThread.start(callback(&watchdogQueue, &EventQueue::dispatch_forever));
```
Comment (cmonr) [inline IdleLoop/main.cpp]: Intendation

File: IdleLoop/main.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#include "mbed.h"
+#include "rtos_idle.h"
+
+DigitalOut led(LED2);
+DigitalOut led1(LED1);
```
Comment (cmonr) [inline IdleLoop/main.cpp]: Two LEDs?

File: IdleLoop/main.cpp (hunk 1 above)
Comment (aashishc1988) [reply IdleLoop/main.cpp]: I think my copy paste to notepadd++ screwed it up. :(

File: IdleLoop/main.cpp (hunk 3 above)
Comment (cmonr) [reply IdleLoop/main.cpp]: I'm gonna nit. Mind updating the names so that they're a bit more descriptive?

At least something like `led2` and `led1`.

Comment (aashishc1988) [reply IdleLoop/main.cpp]: You are right! I will probably add some more comments to this. Working on it

**Comment by aashishc1988** [comment]
added more comments

**Pull request merged**